lib/charms/layer/kubernetes_master.py

```diff
@@ -64,3 +69,25 @@ def get_api_endpoint(relation=None):
         return (ingress_address, STANDARD_API_PORT)
     else:
         return (hookenv.unit_public_ip(), STANDARD_API_PORT)
+
+
+def query_cephfs_enabled(ceph_ep):
+    if not is_flag_set('kubernetes-master.ceph-cli.installed'):
+        check_call(['apt', 'install', '-y', 'ceph-common'])
```

Comment:
`apt update` first?

Reply:
I bring this up because of all the times where `apt install <foo>` has failed for me because the local manifests are referencing a package that no longer exists.

Do charms regularly run `apt update`? If they do, then there's probably no need to call it here. If they don't, then we probably should call it here to be safe and avoid an endless looping failure.

Reply:
layer-basic includes an optimistic-with-fallback helper, so it would probably be worth using that: https://github.com/juju-solutions/layer-basic/blob/master/lib/charms/layer/basic.py#L218-L251